# Day 11 - Exercise 2: Coupon Discount Effectiveness
### Task:
Compare the average units sold per transaction between customers who used a coupon (`Coupon? == 'Yes'`) vs. those who did not (`Coupon? == 'No'`).

### Expected Result:
Does using a coupon encourage purchasing more units?


In [1]:
from pathlib import Path
import pandas as pd

candidates = [Path("Sales.csv"), Path("../Sales.csv"), Path("data/Sales.csv")]
csv_file = next((p for p in candidates if p.exists()), Path("Sales.csv"))
df = pd.read_csv(csv_file)

coupon_stats = df.groupby("Coupon?").agg(
    Avg_Units_Per_Transaction=("Units Sold", "mean"),
    Total_Transactions=("Invoice Number", "count")
)
print("Coupon Usage Comparison:")
print(coupon_stats.round(2))


Coupon Usage Comparison:
         Avg_Units_Per_Transaction  Total_Transactions
Coupon?                                               
No                            1.85                 871
Yes                           1.82                 129


### Insight:
Customers with coupons purchased an average of **1.82 units** vs. **1.85 units** without coupons. Coupons in this dataset drove store foot traffic rather than larger basket quantities!
